bokehjs/src/lib/models/canvas/cartesian_frame.ts

```diff
@@ -207,6 +210,41 @@ export class CartesianFrameView extends StyledElementView {
       return {x: x + offset, y: y + offset}
     }
   }
+
+  readonly position = new InlineStyleSheet()
+
+  override stylesheets(): StyleSheetLike[] {
+    return [...super.stylesheets(), this.position]
+  }
+
+  override rendering_target(): HTMLElement {
+    return this.parent.canvas_view.underlays_el
```

Comment:
```suggestion
    return this.parent.canvas_view.underlays_el ?? this.shadow_el
```

Curious if this could be done. However, I can see this will conflict with the logic in `plot_canvas`, though I don't understand why it does not need `?? this.shadow_el`

Reply:
This would attempt to add an element to it self (`this.shadow_el` is shortcut for `this.el.shadowRoot`). What could work is  `?? this.parent.shadow_el`, but the default (`null`; see `DOMView` base class) is to let the parent to position the child component.

Reply:
I meant `this.parent.shadow_el`. My main point is it seems that (almost) each time this is used, you do a nullish coalescing.

Could `DOMView` be changed to return `this.parent.shadow_el`? I think the responsibility for  `rendering_target` should be isolated to the model itself and not leak out into the code. But could just be I don't understand the nuances of why this is needed.

Reply:
It's the parent that knows where to put its children and that may not be directly into the shadow element. However, sometimes there are multiple options where to put things, e.g. layers on the canvas, thus `rendering_target()` allows the child to make that suggestion.

Reply:
Maybe I just don't see the nuances here. 

I  still think it would be cleaner to `DOMView.rendering_target` to return a shadow element, update the `CartesianFrameView.rendering_target` to be `this.parent.canvas_view.underlays_el ?? this.parent.shadow_el`. And remove the nullish coalescing for the rest of the code. 

This would remove the gocha, which is that you should always do `element_view.rendering_target() ?? this.shadow_el`, when using `rendering_target` or at least need to think about it. 

If you still want to keep it this way, I would like to have an update on the comment that if this returns null, you should use ` ?? this.shadow_el` to always have a target defined.